### DISCUSSION TITLE
How to initialize a GraphQL Java project without using Spring Boot?

### DISCUSSION THREAD

**Discussion by khat33b** [graphql-java/graphql-java#2587] [Q&A]
All the GraphQL samples only use Spring Boot. I want to know how to use GraphQL without using Spring.I want to make a new GraphQL Java project but I do not want to use Spring Boot for it. I want to make the exact equivalent of [this sample project][1] but only using Jersey 3 and Jakarta 5. I have added my Gradle dependenices like this:

    dependencies {
        implementation group: 'org.glassfish.jersey.containers', name: 'jersey-container-servlet', version: '3.0.3'
        implementation group: 'org.glassfish.jersey.inject', name: 'jersey-hk2', version: '3.0.3'
        implementation group: 'org.glassfish.jersey.media', name: 'jersey-media-moxy', version: '3.0.3'
        implementation group: 'jakarta.servlet', name: 'jakarta.servlet-api', version: '5.0.0'
        implementation group: 'jakarta.xml.bind', name: 'jakarta.xml.bind-api', version: '3.0.1'
        implementation group: 'com.sun.xml.bind', name: 'jaxb-impl', version: '3.0.2'
        
        implementation 'com.graphql-java:graphql-java:17.3'
        implementation 'com.graphql-java:graphql-java-extended-scalars:17.0'
        implementation 'com.graphql-java:graphql-java-extended-validation:17.0'
        implementation 'com.google.guava:guava:31.0.1-jre'
        
        implementation 'com.google.code.gson:gson:2.8.8'
    }

I would like to know how to rewrite the [BookDetailsApplication][2], [GraphQLDataFetchers][3] and [GraphQLProvider][4] classes using Jersey 3.

  [1]: https://github.com/graphql-java/tutorials/tree/master/book-details
  [2]: https://github.com/graphql-java/tutorials/blob/master/book-details/src/main/java/com/graphqljava/tutorial/bookdetails/BookDetailsApplication.java
  [3]: https://github.com/graphql-java/tutorials/blob/master/book-details/src/main/java/com/graphqljava/tutorial/bookdetails/GraphQLDataFetchers.java
  [4]: https://github.com/graphql-java/tutorials/blob/master/book-details/src/main/java/com/graphqljava/tutorial/bookdetails/GraphQLProvider.java

**Comment by bbakerman** [comment]
I dont have a lot of time to go into detauls right now but you can

* Have a single`/graphql` entry point in Jersey - one HTTP resource
* Have it create a `GraphQL` instance - this can be per request or you need to use your dependency IoC or statics to make this static one time only
* Create ExecutionInput from the Jersey ars
* Execute stuff with `graphql.execute(executionInput)`
